compiler/passes/splitInit.cpp

```diff
@@ -964,8 +946,33 @@ static bool doFindCopyElisionPoints(Expr* start,
             ++elseIt;
           }
         }
-        // no need to handle leftovers (e.g. ifIt not at end)
-        // because we marked uses above.
+
+        // Migrate remaining local vars in if block to local map.
+        for (; ifIt != ifMap.end(); ++ifIt) {
+          if (cond->thenStmt->contains(ifIt->first->defPoint)) {
+            map[ifIt->first] = ifIt->second;
+          }
+        }
+
+        // Migrate remaining local vars in else block to local map.
+        for (; elseIt != elseMap.end(); ++ elseIt) {
+          if (cond->elseStmt->contains(elseIt->first->defPoint)) {
+            map[elseIt->first] = elseIt->second;
+          }
+        }
```

Comment:
I would think we'd only want to migrate local vars in both ifMap and elseMap to the parent map?